apps/compose_apps.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) 2019 Foundries.io
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import os
+import yaml
+
+from helpers import cmd as cmd_exe, status
+
+
+logger = logging.getLogger(__name__)
+
+
+class ComposeApps:
+    DisabledSuffix = '.disabled'
+
+    class App:
+        DockerComposeTool = 'docker-compose'
+        ComposeFile = 'docker-compose.yml'
+
+        @staticmethod
+        def is_compose_app_dir(app_dir):
+            return os.path.exists(os.path.join(app_dir, ComposeApps.App.ComposeFile))
+
+        def __init__(self, name, app_dir, validate=False):
+            if not self.is_compose_app_dir(app_dir):
+                raise Exception('Compose App dir {} does not contain a compose file {}'
+                                .format(app_dir, self.ComposeFile))
+            self.name = name
+            self.dir = app_dir
+            self.file = os.path.join(self.dir, self.ComposeFile)
+
+            if validate:
+                self.validate()
+
+            with open(self.file) as compose_file:
+                self._desc = yaml.safe_load(compose_file)
+
+        def validate(self):
+            self._run_cmd('config')
+
+        def services(self):
+            return self['services'].items()
+
+        def images(self):
+            for _, service_cfg in self.services():
+                yield service_cfg['image']
+
+        def tag_images(self, tag, image_base_url, allowed_tags=['${TAG}', 'latest']):
```

Comment:
Think you want to default allowed_tags to None and then do an `if allowed_tags is None` in the actual method, or we'll find up with an unexpected python 'bug' some day.

Reply:
It's not relevant in the newest version that I force pushed.